src/main/webapp/app/grading-system/grading-system.component.ts

```diff
@@ -152,8 +152,10 @@ export class GradingSystemComponent implements OnInit {
                 this.dialogErrorSource.next('');
             });
         }
-        this.gradingScale = new GradingScale();
-        this.gradingScale.gradeType = GradeType.GRADE;
+        this.gradingScale = {
```

Comment:
I am not a big fan of this inline object definitions. Can you please use the more object oriented way using an actual constructor and specifying all values?
If gradeSteps is undefined, either give it a default value (empty array) in the class or set this in the constructor or assign it normally (using the `=` operator)